### PR TITLE
Coverage dev

### PR DESCRIPTION
**Pull request by cwoolfo1**
Added action to begin tracking code coverage. Fixed errors in both fdtd.py and pstd.py. Updated unit tests with analytical test cases for the laplacian, gradient, divergence, and curl operators in both fdtd and pstd

### CODE REVIEW

File: PyPIC3D/pstd.py [hunk 1]
```diff
@@ -78,21 +24,20 @@ def spectral_poisson_solve(rho, constants, world):
...
-    k2 = k[0]**2 + k[1]**2 + k[2]**2
-    # calculate the squared wavenumber
-
+    krho = jnp.fft.fftn(rho)
+    # fft the charge density
+    nx, ny, nz = krho.shape
+    kx = jnp.fft.fftfreq(nx, d=dy) * 2 * jnp.pi
+    ky = jnp.fft.fftfreq(ny, d=dz) * 2 * jnp.pi
+    kz = jnp.fft.fftfreq(nz, d=dx) * 2 * jnp.pi
+    kx, ky, kz = jnp.meshgrid(kx, ky, kz, indexing='ij')
```
[the remark this answers is not in the document]
Comment (cwoolfo1) [reply PyPIC3D/pstd.py]: A method has been created to reduce redundancies

File: tests/pstd_test.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    def test_spectral_curl(self):
+        # Use a gradient field, whose curl should be zero
+        Fx = self.gradx
+        Fy = self.grady
+        Fz = self.gradz
+        curlx, curly, curlz = spectral_curl(Fx, Fy, Fz, self.world)
+        maxerrx = jnp.max(jnp.abs(curlx))
+        maxerry = jnp.max(jnp.abs(curly))
+        maxerrz = jnp.max(jnp.abs(curlz))
+        print('Curl (grad field) max abs error:', float(maxerrx), float(maxerry), float(maxerrz))
```
[the remark this answers is not in the document]
Comment (cwoolfo1) [reply tests/pstd_test.py]: Removed old print statements for debugging

File: tests/pstd_test.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        maxerrx = jnp.max(jnp.abs(gradx - self.gradx))
+        maxerry = jnp.max(jnp.abs(grady - self.grady))
+        maxerrz = jnp.max(jnp.abs(gradz - self.gradz))
+        print('Gradient max abs error:', float(maxerrx), float(maxerry), float(maxerrz))
+        self.assertEqual(gradx.shape, self.phi.shape)
+        self.assertEqual(grady.shape, self.phi.shape)
+        self.assertEqual(gradz.shape, self.phi.shape)
+        self.assertTrue(jnp.allclose(gradx, self.gradx, atol=1e-10, rtol=1e-8))
+        self.assertTrue(jnp.allclose(grady, self.grady, atol=1e-10, rtol=1e-8))
+        self.assertTrue(jnp.allclose(gradz, self.gradz, atol=1e-10, rtol=1e-8))
```
[the remark this answers is not in the document]
Comment (cwoolfo1) [reply tests/pstd_test.py]: removed old print statements for debugging
